19.cpp

```diff
@@ -0,0 +1,30 @@
+#include <iostream>
+
+struct ListNode {
+  int val;
+  ListNode* next;
+  ListNode(int x) : val(x), next(NULL) {}
+};
+class Solution {
+ public:
+  ListNode* removeNthFromEnd(ListNode* head, int n) {
+    ListNode* frist_ptr = head;   // found the end node
+    ListNode* second_ptr = head;  // a node before will be deleted node
+    while (n != 0) {
+      --n;
+      frist_ptr = frist_ptr->next;
+    }
+    if (frist_ptr == NULL)
```

Comment:
I think `if (first_ptr)` is clear enough.

Reply:
use this  if (!first_ptr)